executor/executor.cc

```diff
@@ -808,8 +808,8 @@ thread_t* schedule_call(int call_index, int call_num, bool colliding, uint64 cop
 	int i;
 	for (i = 0; i < kMaxThreads; i++) {
 		thread_t* th = &threads[i];
-		if (!th->created)
-			thread_create(th, i);
+		if (!thread_create(th, i))
```

Comment:
If creation of one thread failed, why does it make sense to try to create other ones.
I frankly afraid that you started adding error handing. To do it all properly we would need to add and maintain tons of new code. Even your current code is incomplete: if we create an even, but failed to start the thread, now we need to destroy the even. So we also need to start adding destructors.
If we simply exitf in thread_create as we do now, then we do not need anything of what you added (returning errors, propagation of errors , handling of errors, destructors), right?

Also no printf's. This is not a program that a human operator runs manually once and looks at output. This program is executed gazillions of times and usually nobody cares about output and it's actively harmful because that's additional work. Also pkg/csource needs to deal with that somehow, because we don't want reproducers to print wall of output.
The set of functions that we use in executor: debug, fail, exitf. They have well defined interactions with all other parts of the system.